docs/modules/tiered-storage/pages/overview.adoc

```diff
@@ -35,3 +39,26 @@ Tiered Storage uses the following:
 - Data Persistence and Tiered Storage are mutually exclusive features. If both are enabled simultaneously, `InvalidConfigurationException` is thrown during the member startup.
 
 - SQL is not supported for the Tiered-Storage-backed maps. `UnsupportedOperationException` is thrown when SQL query is executed. To avoid this, use xref:query:predicate-overview.adoc[Predicate API] instead.
+
+== Supported Devices
+
+Hazelcast supports only NVMe (local) SSD devices for Tiered Storage.
+While HDDs and some network storages work with the same file abstraction that local SSDs do, and technically speaking such setups work, the access latency of these devices compared to that of the modern SSDs are so high that using Tiered Storage with such devices are not supported.
+
+== High-Level Design Overview
+
+The Tiered Storage feature for IMaps is backed by a special-purpose, addressable data structure called HybridLog.
+This data structure spans over the memory and the configured disk making the aggregated capacity of the resources available for storing IMap's entries.
+There is a HybridLog data structure for every partition of every IMap.
+
+The IMap entries stored in these HybridLog instances are internally referenced by their logical addresses.
+The key to logical address mapping is done by a hash index component.
+These hash indices are hashtables using separate chaining for conflict resolution.
+The table of these hash indices are stored in the native memory, whereas the chaining information of the buckets are stored on the disk.
+
+Tiered Storage makes sure every accessed IMap entry is available in the memory to build a hot cache of entries.
+This means, if an entry is on the disk, first it gets relocated to the memory inside the HybridLog hosting the entry, and its previous location on the disk becomes unreachable.
+As more and more entries get unreachable, to prevent running out of disk space the disk space occupied by these unreachable entries has to be reclaimed.
```

Comment:
Suggestion: 
-As more and more entries get unreachable, to prevent running out of disk space the disk space occupied by these unreachable entries has to be reclaimed.- 
As more and more entries are relocated to the in-memory region of the HybridLog, the memory space occupied by the unreachable entries on disk are reclaimed.

Reply:
Replaced in 06a2a152059bfb69e1201dfdbf7842c9b65f589f